lib/nexthop.h

```diff
@@ -65,6 +65,11 @@ enum nh_encap_type {
 /* Backup index value is limited */
 #define NEXTHOP_BACKUP_IDX_MAX 255
 
+struct vxlan_nh_encap {
```

Comment:
is mac info completely vxlan-specific - is there a more general evpn use for mac info, for example?

Reply:
I don't understand how a rmac is a property of the nexthop.  It's a property of the l2 information *not* the nexthop.  Why are we storing it in the nexthop?  If the answer if `convenience` then double triple boo

Reply:
this should be a mapping in a different data structure.  I also believe we should move vni's out of nexthops as well.

Reply:
This changes is done based on the current code flow where nexthop store the VNI information. Hence RMAC is also added in the similar lines.

Reply:
Please reread `we should move vni's out of nexthops as well`. it's not the right data abstraction.  continuing down a wrong line does us no good.

Reply:
Isn't part of the issue that the "nexthop" in some of these cases requires this additional information in order to be installed - the normal arp/ND resolution won't work without extra L2 info and processing? what kind of thing would you prefer: an 'info' pointer, instead of directly-embedded L2 info? or some entirely new evpn or L2 object that zebra would manage using a new set of zapi messages?

but if we've ended up here because we've piggy-backed L2/evpn info in the regular route zapi messaging, and then stuffed that info into the nexthop struct for, as you say, convenience, then I agree, it needs to be revised. I don't love the way that evpn/vxlan mac installation occurs currently, as a synchronous side-effect of parsing a zapi nexthop.

Reply:
Sorry our message crossed, so did see your earlier response before my reply.

Reply:
all nexthops need relevant l2 information in order for it to work properly.  Yet we don't carry that information in the nexthop.
A nexthop is L3 reachability.  VNI's RMACS are l2 reachability.  If we stuff this together we are going to have fun with nexthops when a L3 reachability stays live but the l2 changes underneath.  If we *need* to pass on l2 reachability to lower levels than that needs to be a separate action controlled by seperate data structures in the zebra code.  This is the reason they are separate things.

Reply:
@donaldsharp I have one more question. Is it ok to extract the RMAC information in the function netlink_route_info_add_nh using API zl3vni_nh_lookup(zl3vni, nexthop) and send the RMAC along with prefix via FPM channel.

Reply:
I think this merits an actual discussion on how to properly do this in FRR.  I am not sure I have an off the cuff answer here.  I also think that what you proposing is not the right thing to do.  Let me get together w/ Mark/Vivek/Anuradha  and figure it out.

Reply:
Thanks for your response